### PR TITLE
JENA-2015: RDF* annotation syntax

### PR DESCRIPTION
**Pull request by afs**
[JENA-2015](https://issues.apache.org/jira/projects/JENA/issues/JENA-2015)

includes as noted the Javacc turtle parser cleanup.

Does not include companion SPARQL* annotation syntax. (RDF* term syntax is already implemented.)

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/riot/RDFParser.java [hunk 1]
```diff
@@ -334,8 +334,11 @@ private void parseNotUri(StreamRDF destination) {
             throw new RiotException("Failed to determine the RDF syntax (.lang or .base required)");
 
         ReaderRIOT readerRiot = createReader(ct);
-        if ( readerRiot == null )
-            throw new RiotException("No parser registered for content type: " + ct.getContentTypeStr());
+        if ( readerRiot == null ) {
+//            readerRiot = createReader(lang);
+//            if ( readerRiot == null )
+                throw new RiotException("No parser registered for content type: " + ct.getContentTypeStr());
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/RDFParser.java:340]: Format/delete?

File: jena-arq/src/main/java/org/apache/jena/riot/lang/extra/LangParserBase.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    protected final Node nRDFobject     = RDF.Nodes.object ;
+
+    protected StreamRDF stream;
+    protected ParserProfile profile;
+
+    public LangParserBase() { }
+    
+    // These are essential calls unless the parser takes over the functions. 
+    // They can't easily be in the constructor because this class is inherited
+    // by the JaavCC generated parser.
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/lang/extra/LangParserBase.java:66]: s/JaavCC/JavaCC

File: jena-arq/src/main/java/org/apache/jena/riot/lang/extra/LangParserBase.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                    throw new QueryParseException("Bad surrogate pair (end of string)", line, column);
+                char ch1 = string.charAt(i);
+                if ( ! Character.isLowSurrogate(ch1) ) {
+                    throw new QueryParseException("Bad surrogate pair (high surrogate not followed by low surrogate)", line, column);
+                }
+            } else if ( Character.isLowSurrogate(ch) ) {
+                throw new QueryParseException("Bad surrogate pair (low surrogate without high surrogate)", line, column);
+            }
+        }
+    }
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/lang/extra/LangParserBase.java:116]: Was this code elsewhere and it was then moved here? Or is it duplicated? I remember seeing some similar code a few days ago I think.

**Review by kinow — APPROVED** [review]
:+1: looks good to me. Today-I-Learned about that `{| ... |}` syntax. Will try to remember that one, although I prefer to be more explicit and verbose I guess, with the syntax currently tested in this PR.

File: jena-arq/src/main/java/org/apache/jena/riot/lang/extra/LangParserBase.java (hunk 3 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/riot/lang/extra/LangParserBase.java:116]: Duplicated; there is some other methods copied as well.

The tiny difference is that the exceptions are RiotParseException ... except I missed this method!  Changed.

**Comment by afs** [comment]
There are some functions copied - the exceptions are different (QueryParseException vs RiotParseException) - except I missed this function so updated to RiotParseException.

**Pull request merged**